query/src/test/scala/filodb/query/exec/BinaryJoinSetOperatorSpec.scala

```diff
@@ -1369,5 +1370,227 @@ class BinaryJoinSetOperatorSpec extends AnyFunSpec with Matchers with ScalaFutur
+      (Map( "label1".utf8 -> "value1".utf8, "onLabel".utf8 -> "onValue1".utf8)-> Seq((100, 1.0), (200, 2.0), (300, 3.0)))
+    ))
+    val rhsRv = rangeVectors(List(
+      (Map( "label1".utf8 -> "value1".utf8)-> Seq((0, Double.NaN), (10, 1.0), (20, Double.NaN)))
+    ))
+
+    val map = exec.setOpUnless(lhsRv, rhsRv).map( rv => rv.key.labelValues -> rvRowsToListOfTuples(rv)).toMap
+    map.get(Map("label2".utf8 -> "value2".utf8, "onLabel".utf8 -> "onValue1".utf8)) match {
+      case Some(matched)  => assertListEquals(matched, List((0,1.0), (10,Double.NaN), (20,Double.NaN), (30,3.0)))
+      case None           => fail("Expected to find a matching RV for key Map(label2 -> value2, onLabel -> onValue1)")
+    }
+
+    map.get(Map( "label1".utf8 -> "value1".utf8, "onLabel".utf8 -> "onValue1".utf8)) match {
+      case Some(matched)       => matched shouldEqual Seq((100, 1.0), (200, 2.0), (300, 3.0))
+      case None                => fail("Expected to find a matching RV for key Map(label1 -> value1, onLabel -> onValue1)")
+    }
+  }
+
+  it("should perform A - B correctly when on is given") {
+    val exec = SetOperatorExec(QueryContext(), dummyDispatcher, Nil, Nil, BinaryOperator.LUnless, Seq("onLabel"), Nil, "_metric_", None)
+
+
+    val lhsRv = rangeVectors(List(
+      (Map( "label1".utf8 -> "value1".utf8)-> Seq((0, Double.NaN), (10, 1.0), (20, Double.NaN))),
+      (Map( "label2".utf8 -> "value2".utf8, "onLabel".utf8 -> "onValue1".utf8)-> Seq((0, 1.0), (10, 2.0), (20, 3.0))),
+      (Map( "label2".utf8 -> "value2".utf8, "onLabel".utf8 -> "onValue1".utf8)-> Seq((10, 1.0), (20, 2.0), (30, 3.0))),
+      (Map( "label1".utf8 -> "value1".utf8, "onLabel".utf8 -> "onValue1".utf8)-> Seq((100, 1.0), (200, 2.0), (300, 3.0)))
+    ))
+    val rhsRv = rangeVectors(List(
+      (Map( "label1".utf8 -> "value1".utf8)-> Seq((0, Double.NaN), (10, 1.0), (20, Double.NaN)))
+    ))
+
+    val map = exec.setOpUnless(lhsRv, rhsRv).map( rv => rv.key.labelValues -> rvRowsToListOfTuples(rv)).toMap
+    map.get(Map("label2".utf8 -> "value2".utf8, "onLabel".utf8 -> "onValue1".utf8)) match {
+      case Some(matched)  => assertListEquals(matched, List((0,1.0), (10,Double.NaN), (20,Double.NaN), (30,3.0)))
+      case None           => fail("Expected to find a matching RV for key Map(label2 -> value2, onLabel -> onValue1)")
+    }
+
+    map.get(Map( "label1".utf8 -> "value1".utf8, "onLabel".utf8 -> "onValue1".utf8)) match {
+      case Some(matched)       => matched shouldEqual Seq((100, 1.0), (200, 2.0), (300, 3.0))
+      case None                => fail("Expected to find a matching RV for key Map(label1 -> value1, onLabel -> onValue1)")
+    }
+
+  }
+
+
+  it("should perform A - B correctly when on and ignoring is provided is given") {
+    val exec = SetOperatorExec(QueryContext(), dummyDispatcher, Nil, Nil, BinaryOperator.LUnless, Nil, Seq("label1", "label2"), "_metric_", None)
```

Comment:
`on` is unused (the title says otherwise)

Reply:
Will change the description